Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx

```diff
@@ -17,10 +11,8 @@
 
 #ifdef SW_USE_OPENMP
 #include <omp.h>
-const int global_iteration = 1;
-#else /* SW_USE_OPENMP */
-const int global_iteration = 1;
```

Comment:
I am not sure the current state of the OpenMP stuff in ShapeWorks.  But the reason this "global_iteration" const exists is to allow the OpenMP version of the code to run the parallel, per-shape part of the iteration multiple times per global part (e.g. m_GradientFunction->BeforeIteration()).  You can see that part is outside the OpenMP parallel loop, so the program runs in serial for that call.

Back when I added this, I was experimenting with allowing multiple passes of the parallel section.  The result was that the code ran quite a bit faster, but the result is not the same as the serial version.  At some point for safety, I left the checked-in code with "global_iteration = 1" for both cases, and would just change that number to '20' or whatever in my own local branch.

At this point, I have no idea if that would even work.  I have not had a chance to re-evaluate the parallelization opportunities.

Reply:
Makes sense. I just notices that it was getting set in both the if and the else so I moved it outside of the ifelse